visallo/visallo
# Performance changes from http2 branch (3.1.1)

* Production build wasn't calling the minification task, added it back for comment/whitespace removal
* Change RequireJS loader in web worker to allow parallel downloading of worker resources
* Switch as many 3rd party resources to use prebuilt minified versions as possible.

Also prepares for the use of statically compressed web app resources in future build.

Testing Instructions:  
Verify files like services/config.js `/jsc/data/web-worker/services/config.js` doesn't have comments (except for the source map comment in last line)
Verify app works as expected in all supported browsers. Failures would be noticeable right away with files not loading correctly.

CHANGELOG
Changed: Performance improvements to users visiting with empty cache and connections with noticeable latency


## Performance Statistics

Measured to load login screen on enterprise test

|         | Before           | After  |
| ---- |:------------:| -----:|
| Size  | 654kb | 587kb |
| Time (No Latency)  | 5.1s | 3.4s |
| Time (150ms Latency)       | 15.8s      |   8.8s |

### Timeline Recording Before

<img width="1201" alt="timeline-before" src="https://cloud.githubusercontent.com/assets/7098/24719087/0e04a98e-19fe-11e7-8218-fad6b4e75efb.png">

### Timeline Recording After

<img width="1204" alt="screen shot 2017-04-05 at 12 56 59 pm" src="https://cloud.githubusercontent.com/assets/7098/24719445/6785d702-19ff-11e7-9185-5f3775e10cd3.png">
